wireapp/kalium
# fix: secure message draft saving [WPB-11325]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11325" title="WPB-11325" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-11325</a>  [Android] Can not open conversations
  </td></table>
  <br />
 




----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

There were situations when saving message draft cause error in db `android.database.sqlite.SQLiteConstraintException: FOREIGN KEY constraint failed (code 787)` 

### Causes (Optional)

Not working conversations when there was some saved draft

### Solutions

Secure draft saving with checks needed to save a draft

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:


### Testing

#### Test Coverage (Optional)


#### How to Test

1. Start typing message in conversation with some reply or editting message
2. Remove on web message only for you which you are editing or replying to
3. continue typing on android
4. there should not be any issues with db on logs
